core/class/new_spec.rb

```diff
@@ -96,11 +97,10 @@ def message2; "hello"; end
 
   it "raises a TypeError when given a non-Class" do
     error_msg = /superclass must be a Class/
-    -> { Class.new("")         }.should raise_error(TypeError, error_msg)
-    -> { Class.new(1)          }.should raise_error(TypeError, error_msg)
-    -> { Class.new(:symbol)    }.should raise_error(TypeError, error_msg)
-    -> { Class.new(mock('o'))  }.should raise_error(TypeError, error_msg)
-    -> { Class.new(Module.new) }.should raise_error(TypeError, error_msg)
+    test_cases = ["", 1, :symbol, mock('o'), Module.new, BasicObject.new].each do |superclass|
```

Comment:
This makes it more DRY, but also likely makes it impossible to know which one fails if one of them fails.
So for specs, "clarity of what fails if it fails" is often more important than DRY.

Anyway I thin it's overkill to test so many random objects, so I'd just test with `""` and `Module.new`.

Reply:
It already had such issue as the whole test failed on first failure. I'll split it in different test methods using `define_method`.

> Anyway I think it's overkill to test so many random objects, so I'd just test with "" and Module.new.

I disagree, there's no reason not to (performance impact is insignificant) and different Ruby implementations might optimize some special objects making bugs there. Case in point Opal works correctly with `Module.new` but fails with `BasicObject.new` because it's special as it doesn't inherit `Object` making most things not available in it.

Reply:
I think you misunderstood what I meant. Currently, if the `:symbol` example fails, we'll know it's that one from the backtrace line 101 (or from the error message, but that might not always show the incorrect argument).
With that refactor, we won't.

Adding `BasicObject.new` sounds a good idea, I missed that.
I don't mind keeping the other variants either, although some of them do seem redundant.

Reply:
Updated PR. Now it will fail with exact error `raises a TypeError when trying to extend an instance of BasicObject `